### PR TITLE
support stream function

### PR DESCRIPTION
**Pull request by micocube**
When `OpenAI.instance.chat.createStream` passing functions parameters, set stream=True, the first return of stream has no parameters, but you can listen until the end of the stream, and the parameters are spliced together to be the complete parameters，`example: test/fucntion_test.dart#L79`

### CODE REVIEW

**Comment by anasfik** [comment]
Hi, this seems a very big help, it will be reviewed and approved ASAP.

Thank you!

**Comment by Amonc** [comment]
@anasfik @micocube when it will be merged?

**Comment by Josecodesalot** [comment]
Lets merge it! @anasfik  :D

**Comment by davidmigloz** [comment]
If you need streaming with function calling support you can use [this other client](https://pub.dev/packages/openai_dart) while this PR is not merged

**Comment by anasfik** [comment]
Hello @micocube.
I am sorry for delaying this, I laugh at myself by saying:

> Hi, this seems a very big help, it will be reviewed and approved ASAP.

Excuse my laziness, if you have anything else to help with, I will make sure that it will not be another year of wait.

File: lib/src/core/base/chat/interfaces/create.dart [hunk 1]
```diff
@@ -24,6 +24,8 @@ abstract class CreateInterface {
   Stream<OpenAIStreamChatCompletionModel> createStream({
     required String model,
     required List<OpenAIChatCompletionChoiceMessageModel> messages,
+    List<OpenAIFunctionModel>? functions,
+    FunctionCall? functionCall,
     double? temperature,
     double? topP,
```
Comment (anasfik) [inline lib/src/core/base/chat/interfaces/create.dart:30]: a

**Pull request merged**